Sources/TuistHasher/TargetContentHasher.swift

```diff
@@ -96,9 +96,24 @@ public final class TargetContentHasher: TargetContentHashing {
         case let .external(hash: hash): hash
         case .local: nil
         }
+        let settingsHash: String?
+        if let settings = graphTarget.target.settings {
+            settingsHash = try await settingsContentHasher.hash(settings: settings)
+        } else {
+            settingsHash = nil
+        }
+
+        let destinations = graphTarget.target.destinations.map(\.rawValue).sorted()
```

Comment:
This feels like the wrong place since there's implicit knowledge on an incomplete hash. What about moving it to the `projectHash`? It'll require changing the interface of the project type to accept an array of hashes. Alternatively, we can combine the hashes there into a single string.

Reply:
Would you mind expanding on why this is the wrong place? It seems like the right place to me. There's a `projectHash` that's the hash for the whole project and then `TargetContentHasher` combines that with target-specific hashes. I don't see how the newly added hashes differ from what we had with `additionalStrings`.

Reply:
This content hasher has two paths:

1. You don't have a hash, so I calculate it for you.
2. You have a hash, so I don't need to calculate anything for you.

What we are doing here is:

1. You don't have a hash, so I calculate it for you.
2. You have a hash, but that's incomplete.

I think it's hard to reason about, because you don't know it's incomplete unless you trace where that is calculated. I'd move the hash to the target level and have something along the lines of:

```swift
let target = Target(hash: ......")
```

Which is calculated flattening in the package info mapper using the package.resolved information + the settings. 

I think by going from scattered calculation to a more atomic solution will make the solution more resilient.

Reply:
Sounds fair!